### DISCUSSION TITLE
Not working with Docker

### DISCUSSION THREAD

**Discussion by wejobes-abt** [spatie/ray#275] [General]
I am currently trying to get the trial version of Ray to work docker. I have gone through the instructions in the documentation:

 - updated to latest version of Docker, so I can use "host.docker.internal:host-gateway"
 - updated composer to require ray
 - updated docker-compose.yml file to add the extra_hosts setting to my apache container
 - added ray.php to site root with configuration settings for docker (host = host.docker.internal', remote_path = '/var/www/html', local_path = path/in/my/local)
 
 I have confirmed that the ray.php is being loaded by ray, so I know that piece is working. The issue seems to be when ray php attempts to talk to the ray debugger app: it is not getting the information.
 
 Any help would be great. This tool seems very promising and if we can get it working my group of devs will probably be buying licenses.

**Comment by freekmurze** [comment]
I'm hoping one of our Docker users will be able to help you. 

I did already spot a typo in your post: `host.docker.iNternal`

**Reply by wejobes-abt** [reply]
That was a quick typo from my typing. I did not copy paste. Good catch though.

**Reply by jonneroelofs** [reply]
I have it working on a mac. What platform are you on?

**Reply by olivitek** [reply]
Hi @wejobes-abt - I also spent a while figuring out how to get the ray.app to work with Docker. I did everything you've already done, and only had success when I added the following firewall rules in ufw (I'm on Linux mint 20):

![ray-firewall](https://user-images.githubusercontent.com/7258221/109415396-8d139880-7a0c-11eb-8ec9-7d4e45215dd7.png)

You can check if the firewall is the issue by temporarily disabling it.  If setting your own rules, you'll need to check that the subnet is correct by Inspecting your container(s). Look for the actual IP and Gateway. By default, Docker uses 172.17.0.0/16. Mine was different.

Once I configured the firewall it worked :)

**Reply by lode** [reply]
For future reference, some more info was added in https://github.com/spatie/myray.app/pull/33.
